### DISCUSSION TITLE
How can I create a buffer local keymap with Snacks.toggle?

### DISCUSSION THREAD

**Discussion by freddiehaddad** [folke/snacks.nvim#366] [Q&A]
How can I create a buffer local keymap with Snacks.toggle within my lspconfig on attach function? It seems like all the toggle keymaps are global.

**Comment by dpetka2001** [comment]
I believe you could do `Snacks.toggle:map(key, { buffer = 0 })` for buffer-local mappings, since from what I can see from [here](https://github.com/folke/snacks.nvim/blob/98df370703b3c47a297988f3e55ce99628639590/lua/snacks/toggle.lua#L104) the `opts` it accepts are the same as `vim.keymap.set.Opts`, so you should be able to do that.

**Reply by freddiehaddad** [reply]
Hm. I tried that, but it didn't work. I'm doing it within the on_attach function for lspconfig and using `bufnr` instead of `0` (since lspconfig is passing the buffer id as an argument). But when I open another buffer that doesn't have lsp configured, I still see the keybind. It's as if that `buffer = 0` bit isn't being used.

**Reply by dpetka2001** [reply]
No idea, haven't used it myself. Just tried to look at the code of the `map` function.
